src/base/utils/version.h

```diff
@@ -129,6 +129,11 @@ namespace Utils
             return res;
         }
 
+        constexpr bool isValid() const
```

Comment:
`isValid()` in the immutable class? What do you plan to achieve passing errors downstream? In any case I would suggest to write this as `return *this == ThisType{};` to be more straightforward.

Reply:
>isValid() in the immutable class? 

`Version()` ctor is initializing every number to 0, however an all 0 version number can't be valid, or is it? furthermore, I think it could be common to test if a Version instance is valid or not, hence the additional method.
Will do the change.

Reply:
Thanks, I understand you. Well, I would use `std|boost::optional` then, perhaps. Since you incline to a sentinel value, maybe `operator bool()` would be better than, because version is a number (albeit a complex one)? I realise you are tired from this PR already, so I am not insisting in any way.